File: src/14-challenge/solution.js
/**
 * @param {object} workshop - A representation of the workshop
 * @param {string|number|boolean} gift - The gift to find
 * @returns {number[]} The path to the gift
 */
export default function fn(workshop, gift) {
	for (const key in workshop) {
		const value = workshop[key]

		if (typeof value !== 'object') {
			if (value === gift) return [key]
		} else {
			const innerKeyPath = [key, ...fn(value, gift)]
			if (innerKeyPath.length !== 1) return innerKeyPath
		}
	}

	return []
}

File: src/14-challenge/solution.py
from typing import Any, cast


def fn(workshop: dict[str, Any], gift: str | int | bool) -> list[str]:
    for key in workshop:
        value = workshop[key]

        if not isinstance(value, dict):
            if value == gift:
                return [key]
        else:
            inner_key_path = [key, *fn(cast(dict[str, Any], value), gift)]
            if len(inner_key_path) != 1:
                return inner_key_path

    return []
